dpnp/backend/src/dpnp_iterator.hpp

```diff
@@ -351,21 +351,29 @@ class DPNPC_id final
             free_iteration_memory();
             free_output_memory();
 
-            axes = get_validated_axes(__axes, input_shape_size);
+            std::vector<size_type> valid_axes = get_validated_axes(__axes, input_shape_size);
```

Comment:
you create a copy of array here and class member `axis` remains uninitialized.

Reply:
Class member `axis` is initialized in below for-loop.